cpp/open3d/t/geometry/TriangleMesh.h

```diff
@@ -446,6 +446,17 @@ class TriangleMesh : public Geometry, public DrawableGeometry {
     /// \return Rotated TriangleMesh
     TriangleMesh &Rotate(const core::Tensor &R, const core::Tensor &center);
 
+    /// Normalize both triangle normals and vertex normals to length 1.
+    TriangleMesh &NormalizeNormals();
```

Comment:
We might not need kernels for normalizing normals. 
P.S. The goal is to reduce the usage of kernels without taking much performance loss. So, we must always use and improve tensor ops, while avoiding kernel implementation.

The following might work instead of kernel (or might take some fixing):
```cpp
TriangleMesh &TriangleMesh::NormalizeNormals() {
    if (HasVertexNormals()) {
        SetVertexNormals(GetVertexNormals().Contiguous());
        core::Tensor &vertex_normals = GetVertexNormals();
        core::Tensor vertex_normals_norm = vertex_normals.Mul(vertex_normals).Sum({1}).Sqrt();
        vertex_normals = vertex_normals.Div(vertex_normals_norm);
    } else {
        utility::LogWarning("TriangleMesh has no vertex normals.");
    }

    if (HasTriangleNormals()) {
        core::Tensor &triangle_normals = GetTriangleNormals();
        core::Tensor triangle_normals_norm = triangle_normals.Mul(triangle_normals).Sum({1}).Sqrt();
        triangle_normals = triangle_normals.Div(triangle_normals_norm);
    }

    return *this;
}
```

Reply:
The situation when `norm` is 0 will make the normal equal to `nan` if using tensor ops to do `NormalizeNormals`, while we want the normal to be `0` as original. So we may use one more operation like (in python):
```python
triangle_normals[triangle_normals == float('nan')] = 0
```
But currently, the tensor indexing is no very well and this extra operation will cost some time.